e2e-tests/src/synthetic_network.rs

```diff
@@ -0,0 +1,29 @@
+use log::info;
+use synthetic_link::SyntheticNetworkClient;
+
+pub type Milliseconds = u64;
+
+fn create_client(node_name: &str) -> SyntheticNetworkClient {
+    let synthetic_network_url = format!("http://{}:80/qos", node_name);
+    info!("creating an http client for url {}", synthetic_network_url);
+    SyntheticNetworkClient::new(synthetic_network_url)
+}
+
+pub async fn set_out_latency(milliseconds: Milliseconds, node_name: &str) {
```

Comment:
Would either rename this to sth like `create_client_with_latency` or provide the latency as a parameter to the constructor of `SyntheticNetworkClient`.

Reply:
Why? Sounds more confusing for me. I am not creating a client with latency, I am setting latency of the network. This is just a `convenience-method` in spirit of our other methods like this. I can make `SyntheticNetworkClient` public here directly, but I see no point for doing that now.

Reply:
Maybe hand the `client` down as a dependency then? My main gripe here is that you are creating one from scratch in a `set_` function.

Reply:
Not something blocking, though. If you feel strongly about this, resolve the comment.